File: android/src/main/kotlin/app/aglet/rive_mobile/RiveMobilePlugin.kt
package app.aglet.rive_mobile

import androidx.annotation.NonNull
import app.rive.runtime.kotlin.core.Rive

import io.flutter.embedding.engine.plugins.FlutterPlugin

class RiveMobilePlugin: FlutterPlugin {

  override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {

    binding.platformViewRegistry.registerViewFactory(
      "app.aglet/flutter_rive_animation_view", RiveAnimationViewFactory(binding.binaryMessenger)
    )

    Rive.init(binding.applicationContext)
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
  }
}

File: ios/Classes/RiveMobilePlugin.swift
import Flutter
import UIKit

public class RiveMobilePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    registrar.register(RiveAnimationViewFactory(messenger: registrar.messenger(), registrar: registrar), withId: "app.aglet/flutter_rive_animation_view")
  }
}
